Bugfix issue 452:  iOS seek issue at end of book

complacentsee opened the pull request:
This resolves issue #452 failure to seek back at the end of a book. 
I'm less familiar with the podcast side and honestly haven't tested that. I don't think this would cause any issues, but 78d7ba6 could cause issues as it falls back on session details in the event the player doesn't have a track loaded, which is the case after a book ends. 

This does the following:
1. Ensures the UI  shows paused and session current time stops incrementing.
2. Allows a fall back to session time. 
3. Reloads the track queue if it isn't loaded.

complacentsee committed:
Mark session as inactive if end of queue is reached.
Ensures UI shows paused and stops incrementing session time.

complacentsee committed:
Fallback to session details if player has no active track.
Provides better handling if end of book has been reached.

complacentsee committed:
Rebuild track queue while seeking if player does not have a current track.

advplyr commented:
When the item finishes and you press the seek back button it restarts.
Otherwise it works in my testing for both books and podcasts

complacentsee commented:
Restarts meaning it seeks back to the start of the book, or restarts as in restarts playback? Just want to make sure we're seeing the same defect. Local files seem do not show that behavior on my tests. But streaming appears to reset back to the start. 
Looks like 78d7ba6 has more work as on a streaming it's returning nan.
`SEEK: Seek to nan from handler`
I'll keep working on this.

advplyr commented:
Yeah streaming resets back to the start

complacentsee committed:
Fix getCurrentTime for streaming session.
This simplifies logic to  use session current time when session current track start offset is unavailable.

complacentsee commented:
This should resolve the issue. Streaming session vs local session appear return slightly different data, but both store currentTime. So if the player doesn't have a currentTime, we can just use the sessions currentTime directly. Although this does make me wonder why go the bother of any of the logic and not just return the session currentTime.

advplyr commented:
This is working well in my testing, thanks!
